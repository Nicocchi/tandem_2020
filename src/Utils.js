export const shuffle = (arr) => {
    let m = arr.length, t, i;

    // While there remain elements to shuffle
    while (m) {
        // Pick a remaining element
        i = Math.floor(Math.random() * m--);

        // Swap it with the current elment
        t = arr[m];
        arr[m] = arr[i];
        arr[i] = t;
    }

    return arr;
}