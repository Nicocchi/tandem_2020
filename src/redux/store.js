import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import logger from "redux-logger";
import combineReducer from "./reducers";

let store = null;

// Redux dev tools
const composeEnhancers =
    typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
        ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
              // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
          })
        : compose;

const devEnhancer = composeEnhancers(applyMiddleware(thunk, logger));
const enhancer = composeEnhancers(applyMiddleware());

// Create redux store
if (process.env.REACT_APP_NODE_ENV !== "production") {
    store = createStore(combineReducer, devEnhancer);
} else {
    store = createStore(combineReducer, enhancer);
}

export default store;