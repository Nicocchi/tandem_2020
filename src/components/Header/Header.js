import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { endGame } from "./../../redux/actions";

function Header(props) {
    return (
        <div className="game-header">
            {props.isPlaying ? (
                <button data-testid="header-btn" className="answer-btn" onClick={props.endGame}>
                    Quit
                </button>
            ) : null}
        </div>
    );
}

Header.propTypes = {
    endGame: PropTypes.func,
    isPlaying: PropTypes.bool,
};

const mapStateToProps = (state) => {
    return {
        isPlaying: state.gameReducer.isPlaying,
    };
};

export default connect(mapStateToProps, { endGame })(Header);
