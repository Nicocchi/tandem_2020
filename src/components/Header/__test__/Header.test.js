import React from "react";
import ReactDOM from "react-dom";
import Header from "./../Header";
import store from "./../../../redux/store";
import { Provider } from "react-redux";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";
import { stubConsoleError } from "./../../../Utils";

describe("Header", () => {
    afterEach(() => {
        cleanup();
    });

    it("renders without crashing without props", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <Provider store={store}>
                <Header />
            </Provider>,
            div
        );
    });

    it("matches snapshop", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <Header />
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});