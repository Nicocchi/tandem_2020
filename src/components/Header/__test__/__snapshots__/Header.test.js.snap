// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshop 1`] = `
<div
  className="game-header"
/>
`;
